trieste/models/model_interfaces.py

```diff
@@ -274,13 +276,18 @@ def update(self, dataset: Dataset):
         assert dataset.observations.shape[-1] == y.shape[-1]
         data = (dataset.query_points, dataset.observations)
         num_data = data[0].shape[0]
-        num_latent_gps = model.num_latent_gps
+
+        new_q_mu, new_q_var = self.model.predict_f(dataset.query_points, full_cov=True)
+
+        new_q_sqrt = tf.linalg.cholesky(
+            new_q_var
+            + gpflow.config.default_jitter() * (tf.eye(num_data, dtype=new_q_var.dtype)[None])
+        )
+
         model.data = data
         model.num_data = num_data
-        model.q_mu = gpflow.Parameter(np.zeros((num_data, num_latent_gps)))
-        q_sqrt = np.eye(num_data)
-        q_sqrt = np.repeat(q_sqrt[None], num_latent_gps, axis=0)
-        model.q_sqrt = gpflow.Parameter(q_sqrt, transform=gpflow.utilities.triangular())
+        model.q_mu = gpflow.Parameter(new_q_mu)
+        model.q_sqrt = gpflow.Parameter(new_q_sqrt, transform=gpflow.utilities.triangular())
```

Comment:
is this to fix the notebook, or does the implementation nee updating in general?

Reply:
and is there somewhere i can look to understand these changes? a paper? a slack thread?

Reply:
if this was from someone in the pm thread, should they also be a reviewer?

Reply:
not sure how critical this is to the notebook but it needs doing anyway. This is an adaptation of Vincent's code in the bo-demo project. There is no reference paper / thread. Should I ask Vincent for review?

Reply:
yeah that'd be good. i can't be of much help here

Reply:
my only question here would be whether vincent's approach is problem specific, since this class should be quite general. Hopefully he can answer that